numpy/ma/core.py

```diff
@@ -1615,7 +1616,7 @@ def make_mask(m, copy=False, shrink=True, dtype=MaskType):
         else:
             result = np.array(m, dtype=dtype, copy=copy)
     else:
-        result = np.array(filled(m, True), dtype=MaskType)
+        result = np.array(filled(m, True), dtype=dtype)
```

Comment:
I agree, following them all through that is exactly what it is. So, I changed it accordingly.